### PR TITLE
Jasmine Galang

### PR DESCRIPTION
**Pull request by thepotatohead**
@blauente

### CODE REVIEW

File: recipe_batches/recipe_batches.py [hunk 1]
```diff
@@ -2,13 +2,31 @@
...
 import math
 
+
 def recipe_batches(recipe, ingredients):
-  pass 
+    batches = []
+
+    if len(recipe) == len(ingredients):
+        for k in ingredients:
+            batches.append(ingredients[k] // recipe[k])
```
Comment (grantreighard) [inline recipe_batches/recipe_batches.py:11]: This integer division is probably the easiest way to get the answer, you can also do `math.floor()`.

File: recipe_batches/recipe_batches.py [hunk 2]
```diff
@@ -2,13 +2,31 @@
...
+    else:
+        batches.append(0)
+
+    return min(batches)
```
Comment (grantreighard) [inline recipe_batches/recipe_batches.py:15]: Good job getting this to work. I thought you said you didn't get anything passing tests?

File: stock_prices/stock_prices.py [hunk 3]
```diff
@@ -2,14 +2,29 @@
...
 def find_max_profit(prices):
-  pass
+    profit = 0
+
+    for i in range(len(prices) - 1):
+        for j in range(i + 1, len(prices)):
+            if prices[j] - prices[i] > profit:
+                profit = prices[j] - prices[i]
+
+    return profit
```
Comment (grantreighard) [inline stock_prices/stock_prices.py:15]: I also had a double-for loop for my solution. However, you need to account for the negative answer in the test file when all of the "profits" are negative.

File: recipe_batches/recipe_batches.py (hunk 2 above)
Comment (thepotatohead) [reply recipe_batches/recipe_batches.py:15]: I worked through it last night 😄

Comment (grantreighard) [reply recipe_batches/recipe_batches.py:15]: That's awesome. Keep up the good work!

File: climbing_stairs/climbing_stairs.py [hunk 4]
```diff
@@ -2,13 +2,28 @@
...
+    if n == 0 or n == 1:
+        return 1
+
+    if n == 2:
+        return 2
+
+    if n == 3:
+        return 4
+
+    return climbing_stairs(n - 1) + climbing_stairs(n - 2) + climbing_stairs(n - 3)
```
Comment (grantreighard) [inline climbing_stairs/climbing_stairs.py:16]: This is close, you'll also want to implement a cache that returns the cache entry if found, or that creates it otherwise as a result of three recursive calls to the function of `n-1`, `n-2`, and `n-3`. Beej went over this yesterday.
